### PR TITLE
Merges migrate and delete

### PR DESCRIPTION
**Pull request by ebenoist**
This PR looks a *bit* noisier than it is. The last commit has the functional changes.

The 10,000 foot view is:

```
[eb/merge-and-cleanup][~/dev/migr8]❥ bin/migr8
NAME:
   migr8 - It's time to move some redis

USAGE:
   bin/migr8 [global options] command [command options] [arguments...]

VERSION:
   0.0.0

COMMANDS:
   migrate	Migrate one redis to a new redis
   delete
   help, h	Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --source, -s "127.0.0.1:6379"	The redis server to pull data from
   --dest, -d "127.0.0.1:6379"		The destination redis server
   --workers, -w "2"			The count of workers to spin up
   --batch, -b "10"			The batch size
   --prefix, -p 			The key prefix to act on
   --clear-dest, -c			Clear the destination of all it's keys and values
   --help, -h				show help
   --version, -v			print the version
```

I brought in gb has a build tool, [codegansta's cli library](https://github.com/codegangsta/cli), and merged the migrate and delete commands into subcommands. The first commit vendors in our two dependencies.

### CODE REVIEW

File: src/migr8/migrate.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	var response string
+	_, err := fmt.Scanln(&response)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	if response == "Y" {
+		dest_conn := destConnection(dest)
+		fmt.Println("Deleting all keys of destination")
+		_, err := dest_conn.Do("flushall")
```
Comment (kylecrum) [inline src/migr8/migrate.go]: Minor point. I was thinking about this. We should probably do a flushdb instead. Flushall removes all keys from all dbs, which might be a bit unexpected

File: vendor/src/github.com/codegangsta/cli/LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Copyright (C) 2013 Jeremy Saenz
```
Comment (kylecrum) [inline vendor/src/github.com/codegangsta/cli/LICENSE]: :+1:

**Comment by kylecrum** [comment]
:100:

**Comment by skwp** [comment]
Nice, in the future maybe do a separate PR for vendored deps and the actual code change so it's easier to read

**Comment by skwp** [comment]
Also are vendored deps the way to go? itseems like a lotta code in this repo

File: src/migr8/migrate.go (hunk 1 above)
Comment (skwp) [inline src/migr8/migrate.go]: :+1: to that

**Comment by kylecrum** [comment]
@skwp it's the go way, yes.

**Comment by ebenoist** [comment]
@skwp for sure, thats why I put it in a different commit. But I like the idea of a different PR.

And vendored deps are definitely the way to go, but I tend to be in the vendor all the things camp.

**Comment by ebenoist** [comment]
Added some basic testing

**Comment by ebenoist** [comment]
Closing for better review

**Pull request closed without merging**